doc/specifications/rest/REST-API-V2.rst

```diff
@@ -4888,7 +4890,28 @@ List Users
           User_
 
 :Error Codes:
-    :401: If the user has no permission to read users
+    :404: If there are no visibile users matching the filter
```

Comment:
> I'm a bit torn on if the new not found case should be 404 or just a empty user ref list, in case of roleId I would have expected the latter, while in case of unique ids I would have expected the former :P

I've been a bit torn as well. But overall, we don't have so many solutions: if we go for empty list, it means we send a 200, meaning there is no way to distinguish results with HEAD.

I concluded that if you ask for users matching a filter, and none matches, getting responded "not found" kinda makes sense. And it doesn't really break anything.